napari/utils/io.py

```diff
@@ -305,12 +352,121 @@ def read_csv(filename: str) -> Tuple[np.array, List[str]]:
     column_names : list
         List of column names for table data.
     """
-    with open(filename, newline='') as csvfile:
-        reader = csv.reader(csvfile, delimiter=',')
-        output_data = []
-        for row_index, row in enumerate(reader):
-            if row_index == 0:
-                column_names = [str(i) for i in row]
-            else:
-                output_data.append([float(i) for i in row])
-        return np.array(output_data), column_names
+    column_names, *data = iter_csv(filename)
+    return np.array(data), column_names
+
+
+def guess_layer_type_from_column_names(column_names):
+    """Guess layer type based on column names from a csv file.
+
+    Parameters
+    ----------
+    column_names : list of str
+        List of the column names from the csv.
+
+    Returns
+    -------
+    str or None
+        Layer type if recognized, otherwise None.
+    """
+    if column_names[:2] == ['index', 'axis-0']:
+        return 'points'
+    elif column_names[:4] == ['index', 'shape-type', 'vertex-index', 'axis-0']:
+        return 'shapes'
+    else:
+        return None
+
+
+def read_points_csv(filename: str) -> FullLayerData:
+    """Read a points csv file written by the napari builtin.
+
+    Parameters
+    ----------
+    filename : str
+        Filename for reading csv.
+
+    Returns
+    -------
+    data : array
+        Points data.
+    meta : dict
+        Points metadata.
+    layer_type : str
+        Points layer type.
+    """
+    gen = iter_csv(filename)
+    column_names = next(gen)
+
+    layer_type = guess_layer_type_from_column_names(column_names)
+    if layer_type != 'points':
+        raise ValueError('Points csv not recognized')
+
+    table = np.array(list(gen))
+
+    data_axes = [cn.startswith('axis-') for cn in column_names]
+    data = np.array(table[:, data_axes]).astype('float')
+
+    # Add properties to metadata if provided
+    prop_axes = np.logical_not(data_axes)
+    prop_axes[0] = False
+    meta = {}
+    if np.any(prop_axes):
+        meta['properties'] = {}
+        for ind in np.nonzero(prop_axes)[0]:
+            values = table[:, ind]
+            try:
+                values = np.array(values).astype('float')
+            except ValueError:
+                pass
+            meta['properties'][column_names[ind]] = values
+
+    return data, meta, 'points'
+
+
+def read_shapes_csv(filename: str) -> FullLayerData:
+    """Read a shapes csv file written by the napari builtin.
+
+    Parameters
+    ----------
+    filename : str
+        Filename for reading csv.
+
+    Returns
+    -------
+    data : list of array
+        Shapes data.
+    meta : dict
+        Shapes metadata.
+    layer_type : str
+        Shapes layer type.
```

Comment:
```suggestion
```suggestion
    layer_data : tuple
        3-tuple ``(array, dict, str)`` (points data, metadata, 'shapes')
```